src/keystore.cpp

```diff
@@ -11,6 +11,22 @@ bool CKeyStore::AddKey(const CKey &key) {
     return AddKeyPubKey(key, key.GetPubKey());
 }
 
+void CBasicKeyStore::ImplicitlyLearnRelatedKeyScripts(const CPubKey& pubkey)
+{
+    AssertLockHeld(cs_KeyStore);
```

Comment:
In commit "Implicitly know about P2WPKH redeemscripts"

Could you add `assert(mapKeys.count(pubkey.GetID()) || mapWatchKeys.count(pubkey.GetID())`?

The need to add implicit scripts is confusing enough that I think it would help to document assumptions and potentially catch bugs in the implementation adding them.

Reply:
I don't think those asserts are correct for encrypted wallets.

Reply:
Added.